### PR TITLE
tests for iam

### PR DESCRIPTION
**Pull request by elenatorrey**
fixes issue #xx

## Changes in this PR.

- [X] fixes the avg, min, max methods

## Testing this PR.

1. Do X.
2. Do Y.
3. See Z.

### Expected Output.

```
$ rake spec
[...]
```

@osuosl/devs

### CODE REVIEW

**Comment by elenatorrey** [comment]
@Kennric

**Comment by mathuin** [comment]
https://github.com/osuosl/iam/pull/114#issuecomment-230940962

I posted this to Cayleigh before I realized a few of you were working on the same topic, so I'm sharing the content with all of you.

File: util.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    sum / data.length
+  end
+
+  def self.min_val(data)
+    # return the minimum of values
+    return 0 if data.empty?
+    min = data[0][:value]
+    data.each do |item|
+      min = item[:value] if item[:value] < min
+      return min
```
Comment (mathuin) [inline util.rb]: Are you sure this line is in the right place?  Compare it to the following method.

File: util.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+class Util
+  def self.val_id(data)
+    # checks if the hashes are empty
+    # checks if the values of id are fixed numbers
+    return 0 if data.empty?
+    data.each do |item|
+      id = item[:id]
+      return false if id.class != Fixnum
+    end
+    true
```
Comment (mathuin) [inline util.rb]: Is this line necessary?  What happens if you remove the `true` in this method and the method that follows?  Do the methods return the same results?  Which one is more idiomatic?

File: util.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  end
+
+  def self.id_not_neg(data)
+    # checks if the id is not negative
+    data.each do |item|
+      return false if item[:id] <= 0
```
Comment (alxngyn) [inline util.rb]: this function will return false if the item is 0

**Comment by elenatorrey** [comment]
@Kennric @alxngyn

**Comment by alxngyn** [comment]
@elenatorrey there are some still pressing concerns from Mathuin and I

**Comment by elenatorrey** [comment]
Since docker breaks everything I can't fix it right now:(

**Comment by alxngyn** [comment]
@elenatorrey 
Can we fix the merge conflicts and retest? I think we can use your validation tests in the future.

**Pull request closed without merging**